x-pack/plugins/upgrade_assistant/server/lib/apm/index.ts

```diff
@@ -201,6 +201,19 @@ export const apmReindexScript = `
 
       }
 
+      // bump timestamp.us by span.start.us for spans
+      // shouldn't @timestamp this already be a Date?
+      if (ctx._source.processor.event == "span" && context.service.agent.name == "js-base"){
+        def ts = ctx._source.get("@timestamp");
+        if (ts != null && !ctx._source.containsKey("timestamp") && ctx._source.span.start.containsKey("us")) {
+           // add span.start to @timestamp for rum documents v1
+            ctx._source.timestamp = new HashMap();
```

Comment:
Curious why you need to set the HashMap if you're overwriting it a few lines down.

Reply:
I don't see where it is overwritten. Further down `ctx._source.timestamp.us` is set, which I think is fine, as it makes use of the `HashMap`.  
Any chance you meant `ctx._source.@timestamp`? We have two different timestamp fields here, `@timestamp` holds microsecond precision and was introduced first, while `timestamp.us` holds millisecond precision needed for the UI waterfall.

Reply:
You're right - `timestamp` vs `@timestamp`.